docs/hub-tutorials/join-testnet.md

```diff
@@ -4,64 +4,159 @@ order: 4
-:::
+mkdir -p ~/.gaia/cosmovisor/upgrades
+mkdir -p ~/.gaia/cosmovisor/genesis/bin/
+cp $(which gaiad) ~/.gaia/cosmovisor/genesis/bin/
+
+# Verify cosmovisor and gaiad versions are the same.
+cosmovisor version
+
+# Start Cosmovisor
+cosmovisor start
+```
 
-### Software Upgrade
+#### Upgrading
 
-Now it is time to upgrade the software:
+Cosmovisor will continually poll the `$DAEMON_HOME/data/upgrade-info.json` for new upgrade instructions. When an upgrade is ready, node operators can download the new binary and place it under `$DAEMON_HOME/cosmovisor/upgrades/<name>/bin` where `<name>` is the URI-encoded name of the upgrade as specified in the upgrade module plan.
 
-```bash
-git clone https://github.com/cosmos/gaia.git
-cd gaia
-git fetch --all && git checkout master
+For the `Vega` upgrade, this would look like:
+```
+# Prepare Vega upgrade directory
+mkdir -p ~/.gaia/cosmovisor/upgrades/Vega/bin
+
+# Download and install the new binary version.
+cd $HOME/gaia
+git pull
+git checkout v6.0.0-rc3
 make install
+
+# Move the new binary to the Vega upgrade directory
+cp $GOPATH/bin/gaiad ~/.gaia/cosmovisor/upgrades/Vega/bin
 ```
 
-::: tip
-_NOTE_: If you have issues at this step, please check that you have the latest stable version of GO installed.
+If Cosmovisor is already running, there's nothing left to do, otherwise run `cosmovisor start` to start the daemon.
+
+### State Sync
+
+::: warning
+State Sync requires Gaia version [`v6.0.0-rc3`](https://github.com/cosmos/gaia/tree/release/v6.0.0-rc3).
 :::
 
-Note we use `master` here since it contains the latest stable release.
-See the [testnet repo](https://github.com/cosmos/testnets) for details on which version is needed for which testnet, and the [Gaia release page](https://github.com/cosmos/gaia/releases) for details on each release.
+There will need to be additional configuration to enable State Sync on the testnet. State Sync requires setting an initial list of `persistent_peers` to fetch snapshots from. This will change and eventually move to the p2p layer when the Cosmos Hub upgrades to [Tendermint `v0.35`](https://github.com/tendermint/tendermint/issues/6491). For the sake of simplicity, this step is already done in the [Configuration & Setup](#configuration-amp=-setup) section.
+
+Visit a [testnet explorer](https://vega-explorer.hypha.coop/) to get a recent block height and corresponding hash. A node operator can choose any height/hash in the current bonding period, but as the recommended snapshot period is 1000 blocks, it is advised to choose something close to current height - 1000. Set these parameters in the code snippet below `<BLOCK_HEIGHT>` and `<BLOCK_HASH>`
+
+```
+cd $HOME/.gaia/config
+sed -i 's/enable = false/enable = true/' config.toml
+sed -i 's/trust_height = 0/trust_height = <BLOCK_HEIGHT>/' config.toml
+sed -i 's/trust_hash = ""/trust_hash = "<BLOCK_HASH>"/' config.toml
+sed -i 's/rpc_servers = ""/rpc_servers = "134.122.35.247:26657,165.22.235.50:26657' config.toml
```

Comment:
Haven't tested these but can imagine it'll post the same error as on line 52.

Reply:
Let me double check...I adapted this from what I used with state sync on mainnet but I'll reproduce and update